### ISSUE TITLE
NullPointerException on Scope class defined in a separate module

### ISSUE THREAD

**Issue by PenguinDan** [InsertKoinIO/koin-annotations#78]
**Describe the bug**
Essentially, looks like we are receiving a null value when attempting to annotate an object with the `@Scope` annotation where the class value lives in a separate module, please see https://github.com/InsertKoinIO/koin-annotations/blob/3425d682008e8e9311416d8fc3f984214ff5c859/compiler/koin-ksp-compiler/src/jvmMain/kotlin/org/koin/compiler/generator/DefinitionGenerationExt.kt#L111

**To Reproduce**
1. Create a Scope living in one module, for example, `UserSessionScope`
2. Annotate an object in a **different** module with `@Scope(UserSessionScope::class)`
3. Attempt to compile

**Expected behavior**
A `scope<UserSessionScope>` block is generated in the module

**Current Behavior**
The stack trace
```
e: [ksp] java.lang.NullPointerException
	at DefinitionGenerationExtKt.generateScope(DefinitionGenerationExt.kt:111)
	at ModuleGenerationExtKt.generateDefinitions(ModuleGenerationExt.kt:84)
	at ModuleGenerationExtKt.generateClassModule(ModuleGenerationExt.kt:61)
	at org.koin.compiler.generator.KoinGenerator.generateModule(KoinGenerator.kt:63)
	at org.koin.compiler.generator.KoinGenerator.generateModule$default(KoinGenerator.kt:54)
	at org.koin.compiler.generator.KoinGenerator.generateModules(KoinGenerator.kt:42)
	at org.koin.compiler.BuilderProcessor.process(BuilderProcessor.kt:49)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$6$1.invoke(KotlinSymbolProcessingExtension.kt:291)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$6$1.invoke(KotlinSymbolProcessingExtension.kt:289)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.handleException(KotlinSymbolProcessingExtension.kt:394)
	at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:289)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:123)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:99)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:257)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:42)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:115)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:248)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:88)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:47)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:167)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:53)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:101)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:47)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1645)
	at jdk.internal.reflect.GeneratedMethodAccessor104.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(Unknown Source)
	at java.rmi/sun.rmi.transport.Transport$1.run(Unknown Source)
	at java.rmi/sun.rmi.transport.Transport$1.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Unknown Source)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(Unknown Source)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(Unknown Source)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

e: Error occurred in KSP, check log for detail
```

Using koin-annotation 1.2.0

**Comment by PenguinDan** [comment]
Reading into the KSP documentation, could updating the code block from `type.containingFile!!.packageName` to `type.packageName` fix things?

Reading the documentation of `KSDeclaration.packageName` 
```
    /**
     * The name of the package at which this declaration is declared.
     */
    val packageName: KSName
```

**Comment by arnaudgiuliani** [comment]
Yep, good catch 👍 Fixed in 1.2.1 66b84219c3fe670c3a94540cf2dc1e5741ba66df
